### PR TITLE
[ASSIST-411]: Implement Kanban / Trello style board for tasks by Status 

### PR DESCRIPTION
**Pull request by Orrison**
https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/411/

### CODE REVIEW

**Review by mxm1070 — APPROVED** [review]

File: app-modules/task/resources/views/filament/pages/list-tasks.blade.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                            $activeTab = strval($activeTab);
+                            $tabKey = strval($tabKey);
+                        @endphp
+
+                        <x-filament::tabs.item
+                                :active="$activeTab === $tabKey"
+                                :badge="$tab->getBadge()"
+                                :icon="$tab->getIcon()"
+                                :icon-position="$tab->getIconPosition()"
+                                :wire:click="'$set(\'activeTab\', ' . (filled($tabKey) ? ('\'' . $tabKey . '\'') : 'null') . ')'"
```
Comment (danharrin) [inline app-modules/task/resources/views/filament/pages/list-tasks.blade.php:53]: Weird indentation, if we swap out the Blade formatter (for the better one https://github.com/Stillat/blade-parser-typescript/tree/main/formatter) it should handle this

File: app-modules/task/resources/views/filament/pages/list-tasks.blade.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
+@php
+    use Assist\Livewire\Task\TaskKanban;use Filament\Support\Facades\FilamentView;
```
Comment (danharrin) [inline app-modules/task/resources/views/filament/pages/list-tasks.blade.php:2]: Import formatting

File: app-modules/task/resources/views/filament/pages/list-tasks.blade.php [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                                        .iconColor('danger')
+                                        .send()
+                                }
+                            },
+                        });
+                    });
+                },
+            }))
+        })
+    </script>
```
Comment (danharrin) [inline app-modules/task/resources/views/filament/pages/list-tasks.blade.php:116]: I'd advise moving this into a separate script to ensure it doesnt get sent "over the wire" on every network request

File: app-modules/task/src/Actions/UpdateTask.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+use Illuminate\Database\Eloquent\Model;
+
+class UpdateTask
+{
+    public function handle(Task|Model $task, array $data): Task
+    {
+        $data = collect($data);
+
+        /** @var Task $record */
+        $task->fill($data->except('assigned_to')->toArray());
```
Comment (danharrin) [inline app-modules/task/src/Actions/UpdateTask.php:15]: This does feel a little strange, can we not do this in an observer so we can `Task::update()` like normal?

File: tailwind.config.js [hunk 5]
```diff
@@ -13,5 +15,11 @@ export default {
         './resources/views/filament/**/*.blade.php',
         './vendor/filament/**/*.blade.php',
         './vendor/awcodes/filament-tiptap-editor/resources/**/*.blade.php',
+        "./node_modules/flowbite/**/*.js",
+    ],
+    plugins: [
+        forms,
+        typography,
+        // require('flowbite/plugin')
```
Comment (danharrin) [inline tailwind.config.js:23]: Does this need to stay?

**Review by danharrin — CHANGES_REQUESTED** [review]

File: app-modules/task/resources/views/filament/pages/list-tasks.blade.php (hunk 1 above)
Comment (Orrison) [reply app-modules/task/resources/views/filament/pages/list-tasks.blade.php:53]: I believe I have fixed the indentation. But I would consider changing the Blade formatted outside of the scope here. We can create a separate ticket for that.

File: app-modules/task/resources/views/filament/pages/list-tasks.blade.php (hunk 2 above)
Comment (Orrison) [reply app-modules/task/resources/views/filament/pages/list-tasks.blade.php:2]: Done

File: app-modules/task/resources/views/filament/pages/list-tasks.blade.php (hunk 3 above)
Comment (Orrison) [reply app-modules/task/resources/views/filament/pages/list-tasks.blade.php:116]: What's the best way to do that with Livewire/Filament?

File: tailwind.config.js (hunk 5 above)
Comment (Orrison) [reply tailwind.config.js:24]: For now, yes. I added a TODO to fix it later

File: app-modules/task/src/Actions/UpdateTask.php (hunk 4 above)
Comment (Orrison) [reply app-modules/task/src/Actions/UpdateTask.php]: Removed

**Review by danharrin — APPROVED** [review]

**Pull request merged**